### PR TITLE
OStatus contacts are now automatically converted to AP

### PR DESCRIPTION
**Pull request by annando**
```Onepoll``` is called on a regular basis for every contact. So sooner or later all OStatus contacts will be converted to AP (if these contacts are AP ready).

### CODE REVIEW

File: src/Protocol/ActivityPub/Processor.php [hunk 1]
```diff
@@ -298,6 +298,9 @@ public static function followUser($activity)
 		$item = ['author-id' => Contact::getIdForURL($activity['actor']),
 			'author-link' => $activity['actor']];
 
+		// Ensure that the contact has got the right network type
+		self::switchContact($item['author-id']);
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Processor.php]: Shouldn't it be `Receiver::switchContact()`?

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Processor.php]: This is potentially confusing to anyone who isn't you, can you please be abundantly clear about the differences in the method doc blocks? Like `This method do X, if you want to do Y, use the other one instead`.

Comment (annando) [reply src/Protocol/ActivityPub/Processor.php]: This one is a local one. So you cannot mix them.

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Processor.php]: Gotcha. I've seen you expanded the scope of the other one, so I assumed this one was public too.

Comment (annando) [reply src/Protocol/ActivityPub/Processor.php]: I really fought with myself, if I should make a single function or name them differently. But they are only doing **nearly** the same, so mixing them would blow them up.

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Processor.php]: I'm surprised you didn't go with the usual "Add an obscure boolean flag parameter to make the method do two different things"! 😄

Comment (annando) [reply src/Protocol/ActivityPub/Processor.php]: Although you might not believe me, but sometimes old dogs **can** learn new tricks. 😬

**Pull request merged**